ecosystem/sep-0010.md

```diff
@@ -59,9 +59,15 @@ The authentication flow is as follows:
    the transaction isn't malicious.
 1. The **Client** verifies that the transaction is signed by the **Server Account** obtained through discovery flow.
 1. The **Client** verifies that the transaction's first operation is a Manage Data operation that has its:
-   1. Source account set to the **Client Account**
+   1. Source account set to the **Client Account** or the G-zero address
    1. Key set to `<home domain> auth` where the home domain is the **Home Domain**.
    1. Value set to a nonce value.
+1. If the source account is set to the G-zero address, the **Client** verifies that the transaction has an Invoke Host Function operation with:
```

Comment:
This part of the process is a bit broken, because the signed auth is provided up front without the server being the one that is allowed to choose the unique and random parts of what gets signed. SEP-10 is designed so that the server defines the random component, and can later verify that it generated the transaction by simply generating it's own signature, and then hand to the client for it to sign. But the client doesn't need to sign at this point because they need to sign the internal auth. So yeah we need to rethink this.

Reply:
Can we make it so that if the initial request contain a `C...` address as `account`, the initial server response is a payload with the random params that the client will use to assemble the transaction?

It does cause a bigger divergence between `C` and `G/M` SEP-10 which I don't necessarily love.